openTSNE/affinity.py

```diff
@@ -255,9 +255,17 @@ def check_perplexity(self, perplexity):
 def build_knn_index(
     data, method, k, metric, metric_params=None, n_jobs=1, random_state=None
 ):
+    if not sp.issparse(data) and metric in ["angular", "euclidean", "manhattan", "hamming", "dot"]:
```

Comment:
Instead of introducing the distance naming conventions that annoy uses, I would prefer to have them mapped to the distance metrics we have now e.g. add support for _l1_ and _l2_, _angular_ is equivalent to cosine, I believe, and I'm not sure what _dot_ is useful for (assuming it's the dot product).

Reply:
Yes, I believe "angular" is an alias for "cosine" that Annoy uses. I can map "cosine"->"angular" within the Annoy class. Should l1 and l2 be mapped to euclidean and manhattan? The "dot" is a bit weird, but Annoy explicitly supports it, so why not?

Reply:
> Should l1 and l2 be mapped to euclidean and manhattan?

Yes. "taxicab" should also be mapped to manhattan, since this is what pynndescent supports. I'd like to keep consistency as much as possible.

We can also use cosine distance to calculate the pearson correlation, since it's just the cosine on centered data.

It should also be pretty easy to add the `dot` to pynndescent, so that that could also be used for sparse data.

Reply:
OK, I'll add these alias mappings to the PR, but the support for additional metrics (correlation in annoy, dot in pynndescent) can be added later/separately, I think.

Reply:
ok done